Assets/Scripts/InteractLogic.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Interfaces;
+using Player;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+
+[RequireComponent(typeof(IMovable))]
+public class InteractLogic : MonoBehaviour
+{
+    /// <summary>
+    /// These are functions for character to interact with all objects in the game (ex: moving, following, pick up items, open crate, etc..) using raycast and _NavMeshAgent
+    /// (Won't conflict with character attacking enemy or moving on the map)
+    /// </summary>
+    /// 
+
+    [Tooltip("Showing the object player is focusing")]
+    [SerializeField]
+    private Interactable focus;
+
+    [Tooltip("Showing the object target player is interacting")]
+    [SerializeField]
+    private Transform target;
+
+    private IMovable _movable;
```

Comment:
We don't have IMovable anymore, if you pull master to resolve merge conflicts, you'll see. Please resolve merge conflicts